### ISSUE TITLE
Is it possible to calculate the effective receptive field using Gaussian mixture distribution?

### ISSUE THREAD

**Issue by tylorswfit** [Chasel-Tsui/mmdet-rfla#38]
Is it possible to calculate the effective receptive field using Gaussian mixture distribution?
Is it more accurate to effectively feel the field?

**Comment by Chasel-Tsui** [comment]
Thanks for your insightful comments! 
(1) Theoretically, GMM is more accurate when estimating the distribution, but it is worth noting that according to the paper "Understanding the Effective Receptive Field in Deep Convolutional Neural Networks, NIPS, 2016", the receptive field of each feature point follows single Gaussian distribution, hence I can not expect much improvement if we use GMM to estimate the RF of each feature point. Moreover, it may be hard to determine the parameters of GMM to fit the feature point.
(2) Regarding to the accurate estimation of ERF, I believe it is a direction worth studying. A better way compared to the GMM may be dynamically estimating the ERF during the training process since the ERF gradually shrinks during the network training. If someone can dynamically estimate the ERF for object detection label assignment, this will be a pretty nice work which incorporates the explanability into the learning process of an object detector.
